### PR TITLE
Support for scalars in statistics collector

### PR DESCRIPTION
**Pull request by kawakami-masaki0**
## Pull Request Description:
This PR addresses the following https://github.com/SonySemiconductorSolutions/mct-model-optimization/issues/1451
In PyTorch, if the number of dimensions of the activation tensor is less than 2, an error occured in the statistics collection process.

[Cause]
1. When converting to a Numpy array, the scalar remains 0-dimensional, so the tensor's dimension `x.shape` cannot be obtained.
2. The default value of `out_channel_axis` is 1, which means that axes cannot be set correctly for tensors with dimensions less than 2.

[Solution]
1. Support for scalars in `torch_tensor_to_numpy`.
2. Modified `out_channel_axis` to -1 if the number of dimensions of the activation tensor < 2.

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: tests_pytest/pytorch_tests/unit_tests/core/test_create_stats_collector_for_node.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    pass
+
+class DummyLayer:
+    pass
+
+@pytest.fixture
+def fw_info_mock():
+    fw_info = Mock()
+    fw_info.out_channel_axis_mapping = DefaultDict({Conv2D: 1, Linear: -1, ConvTranspose2d: 1}, 1)
+    return fw_info
```
Comment (kkawa14) [inline tests_pytest/pytorch_tests/unit_tests/core/test_create_stats_collector_for_node.py:37]: It looks like out_channel_axis_mapping is only being tested with PyTorch, 
but from a coverage perspective we should also verify the Keras pattern.
If there are no issues there, we don’t need integration or end-to-end tests.

Comment (kawakami-masaki0) [reply tests_pytest/pytorch_tests/unit_tests/core/test_create_stats_collector_for_node.py:37]: Added test for Keras.
We've confirmed that the out_channel_axis remains -1, so we don't need to create integration or end-to-end tests.
